src/mlpack/methods/ann/layer/instance_norm.hpp

```diff
@@ -0,0 +1,214 @@
+               const double momentum = 0.1);
+
+  /**
+   * Forward pass of the Instance Normalization layer. Transforms the input data
+   * into zero mean and unit variance, scales the data by a factor gamma and
+   * shifts it by beta.
+   *
+   * @param input Input data for the layer
+   * @param output Resulting output activations.
+   */
+  template<typename eT>
+  void Forward(const arma::Mat<eT>& input, arma::Mat<eT>& output);
+
+  /**
+   * Backward pass through the layer.
+   *
+   * @param input The input activations
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+  template<typename eT>
+  void Backward(const arma::Mat<eT>& input,
+                const arma::Mat<eT>& gy,
+                arma::Mat<eT>& g);
+
+  /**
+   * Calculate the gradient using the output delta and the input activations.
+   *
+   * @param input The input activations
+   * @param error The calculated error
+   * @param gradient The calculated gradient.
+   */
+  template<typename eT>
+  void Gradient(const arma::Mat<eT>& input,
+                const arma::Mat<eT>& error,
+                arma::Mat<eT>& gradient);
+
+  //! Get the parameters.
+  OutputDataType const& Parameters() const { return batchNorm.Parameters(); }
+  //! Modify the parameters.
+  OutputDataType& Parameters() { return batchNorm.Parameters(); }
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const
+  { return batchNorm.OutputParameter(); }
+
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return batchNorm.OutputParameter(); }
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return batchNorm.Delta(); }
+  //! Modify the delta.
+  OutputDataType& Delta() { return batchNorm.Delta(); }
+
+  //! Get the gradient.
+  OutputDataType const& Gradient() const { return batchNorm.Gradient(); }
+  //! Modify the gradient.
+  OutputDataType& Gradient() { return batchNorm.Gradient(); }
+
+  //! Get the value of deterministic parameter.
+  bool Deterministic() const { return deterministic; }
+  //! Modify the value of deterministic parameter.
+  bool& Deterministic() { return deterministic; }
+
+  //! Get the mean over the training data.
+  OutputDataType const& TrainingMean() const { return runningMean; }
+  //! Modify the mean over the training data.
+  OutputDataType& TrainingMean() { return runningMean; }
+
+  //! Get the variance over the training data.
+  OutputDataType const& TrainingVariance() const { return runningVariance; }
+  //! Modify the variance over the training data.
+  OutputDataType& TrainingVariance() { return runningVariance; }
+
+  //! Get the number of input units / channels.
+  size_t InputSize() const { return size; }
+
+  //! Get the epsilon value.
+  double Epsilon() const { return eps; }
+
+  //! Get the momentum value.
+  double Momentum() const { return momentum; }
+
+  //! Get the average parameter.
+  bool Average() const { return average; }
```

Comment:
Should we implement a modifier for the parameters as well?

Reply:
`eps`, `momentum` & `average` are declared are const variables & hence no modifier has been implemented.